lms/static/scripts/frontend_apps/components/ContentSelector.js

```diff
@@ -95,23 +99,39 @@ export default function ContentSelector({
   };
 
   /** @param {File} file */
-  const selectLMSFile = file => {
+  const selectCanvasSFile = file => {
     cancelDialog();
     onSelectContent({ type: 'file', file });
   };
 
+  /** @param {File} file */
+  const selectBlackBoardFile = file => {
```

Comment:
This is the newest piece that was added since the draft which adds the `document_url` param by passing up a `URLContent` object to FilePickerApp, and then that of course goes to FilePickerFormFields . The draft version never fully worked because it was not passing the hidden `document_url` parm.

Reply:
To elaborate on this comment to check my understanding:

- When the `FilePickerApp` form is submitted, the frontend indicates to Blackboard the chosen URL in a hidden `document_url` form field rendered by `FilePickerFormFields`
- When the user selects a Blackboard file, this is currently indicated by using a `blackboard://content-resource/{file_id}` URL, rather than a separate form field
- Therefore in order to render the correct form fields we need to separate the code paths for Canvas and Blackboard files here. For Blackboard files we behave as if the user selected a URL, but instead set the URL to the `file.id` value provided by the backend, which is already a `blackboard://content-resource/{file_id}` URL